web/modules/custom/joinup_communities/tallinn/src/Form/DashboardPermissionsForm.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\tallinn\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Site\Settings;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Administrative form for setting the dashboard data permissions.
+ *
+ * Even this is a form designed to update a config value, we're not using
+ * ConfigFormBase because that form is subject to 'config_readonly' form
+ * disabling.
+ */
+class DashboardPermissionsForm extends FormBase {
+
+  /**
+   * If the 'config_readonly' module is installed.
+   *
+   * @var bool
+   */
+  protected $moduleConfigReadonlyIsInstalled;
+
+  /**
+   * Creates a new form object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The factory for configuration objects.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler service.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, ModuleHandlerInterface $module_handler) {
+    $this->setConfigFactory($config_factory);
+    $this->moduleConfigReadonlyIsInstalled = $module_handler->moduleExists('config_readonly');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('module_handler')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'tallinn.dashboard_permissions';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    return [
+      'access_type' => [
+        '#type' => 'radios',
+        '#title' => $this->t('Access to the dashboard data'),
+        '#options' => [
+          'public' => $this->t('Public'),
+          'restricted' => $this->t('Restricted (only moderators and Tallinn collection facilitators)'),
+        ],
+        '#default_value' => $this->config('tallinn.settings')->get('dashboard.access_type'),
+      ],
+      'actions' => [
+        '#type' => 'actions',
+        'submit' => [
+          '#type' => 'submit',
+          '#value' => $this->t('Save configuration'),
+          '#button_type' => 'primary',
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    $this->toggleConfigReadonly(FALSE);
+    $this->configFactory->getEditable('tallinn.settings')
+      ->set('dashboard.access_type', $form_state->getValue('access_type'))
+      ->save();
+    $this->toggleConfigReadonly(TRUE);
```

Comment:
I think this is an inventive solution to give some real time control about the site configuration to moderators, but unfortunately this is not sufficient. On the next deployment the original config as defined in `tallinn.settings.yml` will be reapplied and the access will be restricted again.

If this would be something that needs to be toggled repeatedly and with great urgency then we could come up with a way to exclude certain config keys from being overwritten during deploy. For example we could use the config_filter module to exclude the `tallinn.settings.dashboard.access_type` key.

But I think this is overkill, this will only be toggled one single time, and we know in advance when this will happen so we can schedule a release on that day.

So maybe this whole form is unneeded?

Reply:
Finally, I brought on the table an old idea on how to trick config readonly. I switched to a state entry. Yes, I know, smells a lot as `variable_[get|set]()` :)

Reply:
I had a discussion with Marco and the functional team about this and they explained that they do want this as a toggle in the UI since they don't want to risk the publication of the dashboard being delayed because of a potential deployment issue.

This is a good temporary solution but we should switch to config anyway once the dashboard is public, because state is intended to be temporary. I created a followup to deal with this in September: https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-4662